ql/src/test/results/clientpositive/perf/spark/query80.q.out

```diff
@@ -782,7 +782,7 @@ STAGE PLANS:
                     1 Map 33
                   Statistics: Num rows: 231917759 Data size: 31534108438 Basic stats: COMPLETE Column stats: NONE
                   Select Operator
-                    expressions: _col15 (type: string), _col5 (type: decimal(7,2)), CASE WHEN (_col9 is not null) THEN (_col9) ELSE (0) END (type: decimal(12,2)), (_col6 - CASE WHEN (_col10 is not null) THEN (_col10) ELSE (0) END) (type: decimal(13,2))
+                    expressions: _col15 (type: string), _col5 (type: decimal(7,2)), CASE WHEN (_col9 is not null) THEN (_col9) ELSE (CAST( 0 AS decimal(7,2))) END (type: decimal(7,2)), (_col6 - CASE WHEN (_col10 is not null) THEN (_col10) ELSE (CAST( 0 AS decimal(7,2))) END) (type: decimal(8,2))
```

Comment:
Interesting case change, why is it not 12,2 or 13,2?

Reply:
Also, inside it is 7,2, outside it is 8,2?

Reply:
change to 7,2 is due to the `_col9`'s type which is decimal(7,2). The whole expression's type is determined to be decimal(8,2) for some reason, but this change is in line with previous logic where even though internal type was decimal(12,2) whole case expr was decimal(13,2). In short type inference looks to be more tight now.